Welcome to Data Carpentry Genomics Workshop at University of Nevada, Reno!
===

This website contains materials for the [Data Carpentry Genomics Workshop at University of Nevada, Reno](https://sateeshperi.github.io/2019-01-15-reno/) two-day course (9:00 am-5:00pm), held on 15-16th January, 2019 at the [Mathewson-IGT Knowledge Center - 107](https://library.unr.edu/).

We are very excited to have you here and hope that we can get through a lot of material by the end of the workshop.

This workshop runs under a [Code of Conduct](code_of_conduct.html). Please respect it and be excellent to each other!

Twitter hash tag: #DCUNR 

Notes from the course: 

[Hackmd.io]()

## Instructors

* [Elias Oziolor](https://oziolor.wordpress.com/)
* [Sateesh Peri](https://sateeshperi.github.io/)

This workshop incorporates materials from the [Analysis of Next Generation Sequencing or ANGUS](https://angus.readthedocs.io/en/2018/) and several nonmodel RNAseq workshops. We recommend you visit and explore these materials! ANGUS materials are taught at the [Data Intensive Biology Summer Institute or DIBSI](http://ivory.idyll.org/dibsi/) coordinated at UC Davis every summer by [Dr. Titus Brown](http://ivory.idyll.org/lab), which Elias and Sateesh assist with in their capacity as certified [Carpentry Instructors](https://software-carpentry.org/blog/2017/09/merger.html).

## Schedule in brief

**Day-1**

* 8:00-9:00am [Introduction](https://setac-omics.readthedocs.io/en/latest/intro.html) and RNASeq overview
* 9:00-10:00am [Introduction to CyVerse](https://setac-omics.readthedocs.io/en/latest/jetstream/boot.html)
  * [logging in with a private key](https://setac-omics.readthedocs.io/en/latest/jetstream/login.html)
  * [working with bioconda](https://setac-omics.readthedocs.io/en/latest/jetstream-bioconda-config.html)
* 10:00-10:15am Coffee and sugar
* 10:15-11:00am [Bash](https://setac-omics.readthedocs.io/en/latest/bash_lesson.html)
* 11:00am-12:00pm [Trimming sequencing data](https://setac-omics.readthedocs.io/en/latest/quality-trimming.html)
* 12:00-1:30pm Lunch
* 1:30-2:45pm [Transcriptome assembly](https://setac-omics.readthedocs.io/en/latest/transcriptome-assembly.html)
* 2:45-3:00pm Coffee and sugar
* 3:00-4:00pm [Read quantification](https://setac-omics.readthedocs.io/en/latest/rnaseq-quant.html)
* 4:00-5:00pm [Differential expression](https://setac-omics.readthedocs.io/en/latest/DE.html)

**Day-2**

* 8:00-9:00am [Introduction](https://setac-omics.readthedocs.io/en/latest/intro.html) and RNASeq overview
* 9:00-10:00am [Jetstream lesson](https://setac-omics.readthedocs.io/en/latest/jetstream/boot.html)
  * [logging in with a private key](https://setac-omics.readthedocs.io/en/latest/jetstream/login.html)
  * [working with bioconda](https://setac-omics.readthedocs.io/en/latest/jetstream-bioconda-config.html)
* 10:00-10:15am Coffee and sugar
* 10:15-11:00am [Bash](https://setac-omics.readthedocs.io/en/latest/bash_lesson.html)
* 11:00am-12:00pm [Trimming sequencing data](https://setac-omics.readthedocs.io/en/latest/quality-trimming.html)
* 12:00-1:30pm Lunch
* 1:30-2:45pm [Transcriptome assembly](https://setac-omics.readthedocs.io/en/latest/transcriptome-assembly.html)
* 2:45-3:00pm Coffee and sugar
* 3:00-4:00pm [Read quantification](https://setac-omics.readthedocs.io/en/latest/rnaseq-quant.html)
* 4:00-5:00pm [Differential expression](https://setac-omics.readthedocs.io/en/latest/DE.html)

## Other info

* These lessons are based on the original [Eel Pond Protocol](https://khmer-protocols.readthedocs.io/en/ctb/mrnaseq/index.html), then [edited in 2016](https://eel-pond.readthedocs.io/en/latest/) and [automated](https://github.com/dib-lab/eelpond) by Dr. Tessa Pierce.
* Two other nonmodel RNAseq workshops: [DIBSI 2017](https://dibsi-rnaseq.readthedocs.io/en/latest/) and [SIO-BUG 2017](https://rnaseq-workshop-2017.readthedocs.io/en/latest/index.html)
* [Backup files](backup.html), if something goes wrong
* If you find that you would like more practice working on the command line, here is a [command-line bootcamp](http://rik.smith-unna.com/command_line_bootcamp/?id=yk822u2rpo). In addition, BASH training videos are freely available through the [NH-INBRE program, the UNH Bioinformatics Core facility](http://nhinbre.org/bioinformatics-modules/).
* Source documents for these lessons can be found [on GitHub](https://github.com/WhiteheadLab/2018-setacna-rnaseq/).
