

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  
  <!-- Licensed under the Apache 2.0 License -->
  <link rel="stylesheet" type="text/css" href="_static/fonts/open-sans/stylesheet.css" />
  <!-- Licensed under the SIL Open Font License -->
  <link rel="stylesheet" type="text/css" href="_static/fonts/source-serif-pro/source-serif-pro.css" />
  <link rel="stylesheet" type="text/css" href="_static/css/bootstrap.min.css" />
  <link rel="stylesheet" type="text/css" href="_static/css/bootstrap-theme.min.css" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
    <title>4. Jetstream: install bioconda on a blank Linux machine. &#8212; 2018-setacna-rnaseq 0.0.1 documentation</title>
    <link rel="stylesheet" href="_static/guzzle.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="3. Adding password to a Jetstream instance" href="jetstream/ssh_changepassword.html" />
  
   

  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="jetstream/ssh_changepassword.html" title="3. Adding password to a Jetstream instance"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">2018-setacna-rnaseq 0.0.1 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="container-wrapper">

      <div id="mobile-toggle">
        <a href="#"><span class="glyphicon glyphicon-align-justify" aria-hidden="true"></span></a>
      </div>
  <div id="left-column">
    <div class="sphinxsidebar">
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <h2>Contents</h2>
    <div class="sidebar-localtoc">
      <ul>
<li><a class="reference internal" href="#">4. Jetstream: install bioconda on a blank Linux machine.</a><ul>
<li><a class="reference internal" href="#booting-a-blank-machine">4.1. Booting a “blank” machine</a></li>
<li><a class="reference internal" href="#what-is-bioconda">4.2. What is bioconda?</a></li>
<li><a class="reference internal" href="#what-problems-does-conda-and-therefore-bioconda-solve">4.3. What problems does conda (and therefore bioconda) solve?</a></li>
<li><a class="reference internal" href="#installing-conda-and-enabling-bioconda">4.4. Installing conda and enabling bioconda</a></li>
<li><a class="reference internal" href="#using-conda">4.5. Using conda</a></li>
<li><a class="reference internal" href="#using-bioconda">4.6. Using bioconda</a><ul>
<li><a class="reference internal" href="#finding-bioconda-packages">4.6.1. Finding bioconda packages</a></li>
<li><a class="reference internal" href="#freezing-an-environment">4.6.2. Freezing an environment</a></li>
<li><a class="reference internal" href="#constructing-and-using-multiple-environments">4.6.3. Constructing and using multiple environments</a></li>
<li><a class="reference internal" href="#meditations-on-reproducibility-and-provenance">4.6.4. Meditations on reproducibility and provenance</a></li>
<li><a class="reference internal" href="#using-it-on-your-own-compute-system-laptop-or-hpc">4.6.5. Using it on your own compute system (laptop or HPC)</a></li>
</ul>
</li>
<li><a class="reference internal" href="#bonus-installing-rstudio-web-on-your-running-linux-box">4.7. Bonus: installing RStudio Web on your running Linux box.</a></li>
</ul>
</li>
</ul>

    </div>
  </div>
</div>
  <h4>Previous topic</h4>
  <p class="topless"><a href="jetstream/ssh_changepassword.html"
                        title="previous chapter">3. Adding password to a Jetstream instance</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/jetstream-bioconda-config.md.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <div id="main-search">
      <form class="form-inline" action="search.html" method="GET" role="form">
        <div class="input-group">
          <input name="q" type="text" class="form-control" placeholder="Search...">
        </div>
        <input type="hidden" name="check_keywords" value="yes" />
        <input type="hidden" name="area" value="default" />
      </form>
    </div>
  </div>
</div>
      
    </div>
  </div>
        <div id="right-column">
          
          <div role="navigation" aria-label="breadcrumbs navigation">
            <ol class="breadcrumb">
              <li><a href="index.html">Docs</a></li>
              
              <li>4. Jetstream: install bioconda on a blank Linux machine.</li>
            </ol>
          </div>
          
          <div class="document clearer body">
            
  <div class="section" id="jetstream-install-bioconda-on-a-blank-linux-machine">
<span id="jetstream-install-bioconda-on-a-blank-linux-machine"></span><h1>4. Jetstream: install bioconda on a blank Linux machine.<a class="headerlink" href="#jetstream-install-bioconda-on-a-blank-linux-machine" title="Permalink to this headline">¶</a></h1>
<p>Learning objections:</p>
<ul class="simple">
<li>learn what bioconda is and how to install it</li>
<li>understand basic <code class="docutils literal notranslate"><span class="pre">conda</span></code> commands</li>
<li>learn how to list installed software packages</li>
<li>learn how to manage multiple installation environments</li>
</ul>
<div class="section" id="booting-a-blank-machine">
<span id="booting-a-blank-machine"></span><h2>4.1. Booting a “blank” machine<a class="headerlink" href="#booting-a-blank-machine" title="Permalink to this headline">¶</a></h2>
<p>Follow <a class="reference external" href="https://angus.readthedocs.io/en/2018/jetstream/boot.html">ANGUS instructions, with m1.medium</a>, using “18.04 Ubuntu devel and docker” as the starting image you select – rather than “DIBSI 2018 workshop image”.</p>
<p>Log in via the <a class="reference external" href="https://angus.readthedocs.io/en/2018/jetstream/boot.html#click-on-your-new-instance-to-get-more-information">Web shell or through ssh in your terminal</a> if you are comfortable with that way now.</p>
<p>Note that neither RStudio nor conda are installed.</p>
</div>
<div class="section" id="what-is-bioconda">
<span id="what-is-bioconda"></span><h2>4.2. What is bioconda?<a class="headerlink" href="#what-is-bioconda" title="Permalink to this headline">¶</a></h2>
<p>See <a class="reference external" href="https://www.biorxiv.org/content/early/2017/10/27/207092">the bioconda paper</a> and the <a class="reference external" href="http://bioconda.github.io">bioconda web site</a>.</p>
<p>Bioconda is a community-enabled repository of 3,000+ bioinformatics packages, installable via the <code class="docutils literal notranslate"><span class="pre">conda</span></code> package
manager.  It consists of a set of recipes, <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/blob/master/recipes/sourmash/meta.yaml">like this one, for sourmash</a>, that are maintained by the community.</p>
<p>It just works, and it’s effin’ magic!!</p>
</div>
<div class="section" id="what-problems-does-conda-and-therefore-bioconda-solve">
<span id="what-problems-does-conda-and-therefore-bioconda-solve"></span><h2>4.3. What problems does conda (and therefore bioconda) solve?<a class="headerlink" href="#what-problems-does-conda-and-therefore-bioconda-solve" title="Permalink to this headline">¶</a></h2>
<p>Conda tracks installed packages and their versions.</p>
<p>Conda makes sure that different installed packages don’t have
conflicting dependencies (we’ll explain this below).</p>
</div>
<div class="section" id="installing-conda-and-enabling-bioconda">
<span id="installing-conda-and-enabling-bioconda"></span><h2>4.4. Installing conda and enabling bioconda<a class="headerlink" href="#installing-conda-and-enabling-bioconda" title="Permalink to this headline">¶</a></h2>
<p>Download and install conda:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">curl</span> <span class="o">-</span><span class="n">O</span> <span class="o">-</span><span class="n">L</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">repo</span><span class="o">.</span><span class="n">continuum</span><span class="o">.</span><span class="n">io</span><span class="o">/</span><span class="n">miniconda</span><span class="o">/</span><span class="n">Miniconda3</span><span class="o">-</span><span class="n">latest</span><span class="o">-</span><span class="n">Linux</span><span class="o">-</span><span class="n">x86_64</span><span class="o">.</span><span class="n">sh</span>
<span class="n">bash</span> <span class="n">Miniconda3</span><span class="o">-</span><span class="n">latest</span><span class="o">-</span><span class="n">Linux</span><span class="o">-</span><span class="n">x86_64</span><span class="o">.</span><span class="n">sh</span>
</pre></div>
</div>
<p>Say “yes” to everything the installer asks, and accept default locations by pressing enter when it says “Miniconda3 will now be installed into this location”. (If the prompt looks like this “&gt;&gt;&gt;”, then you are still within the installation process.)</p>
<p>When the installation is complete and the regular prompt returns, run the following command (or start a new terminal session) in order to activate the conda environment:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">source</span> <span class="o">~/.</span><span class="n">bashrc</span>
</pre></div>
</div>
<p>Next, enable various “channels” for software install, including bioconda:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">config</span> <span class="o">--</span><span class="n">add</span> <span class="n">channels</span> <span class="n">defaults</span>
<span class="n">conda</span> <span class="n">config</span> <span class="o">--</span><span class="n">add</span> <span class="n">channels</span> <span class="n">conda</span><span class="o">-</span><span class="n">forge</span>
<span class="n">conda</span> <span class="n">config</span> <span class="o">--</span><span class="n">add</span> <span class="n">channels</span> <span class="n">bioconda</span>
</pre></div>
</div>
<p>Try installing something:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">install</span> <span class="n">sourmash</span>
</pre></div>
</div>
<p>and running it –</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">sourmash</span>
</pre></div>
</div>
<p>will produce some output. (We’ll tell you more about sourmash later.)</p>
<p>yay!</p>
</div>
<div class="section" id="using-conda">
<span id="using-conda"></span><h2>4.5. Using conda<a class="headerlink" href="#using-conda" title="Permalink to this headline">¶</a></h2>
<p>Conda is a “package manager” or software installer. See <a class="reference external" href="https://conda.io/docs/commands.html">the full list of commands</a>.</p>
<p><code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">install</span></code> to install a package.</p>
<p><code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">list</span></code> to list installed packages.</p>
<p><code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">search</span></code> to search packages. Note that you’ll see one package for every version of the software and for every version of Python (e.g. <code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">search</span> <span class="pre">sourmash</span></code>).</p>
</div>
<div class="section" id="using-bioconda">
<span id="using-bioconda"></span><h2>4.6. Using bioconda<a class="headerlink" href="#using-bioconda" title="Permalink to this headline">¶</a></h2>
<p>bioconda is a channel for conda, which just means that you
can “add” it to conda as a source of packages. That’s what the <code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">config</span></code> above does.</p>
<p>Note, Bioconda supports only 64-bit Linux and Mac OSX.</p>
<p>You can check out <a class="reference external" href="https://bioconda.github.io/">the bioconda site</a>.</p>
<div class="section" id="finding-bioconda-packages">
<span id="finding-bioconda-packages"></span><h3>4.6.1. Finding bioconda packages<a class="headerlink" href="#finding-bioconda-packages" title="Permalink to this headline">¶</a></h3>
<p>You can use <code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">search</span></code>, or you can use google, or you can go visit <a class="reference external" href="https://bioconda.github.io/recipes.html#recipes">the list of recipes</a>.</p>
</div>
<div class="section" id="freezing-an-environment">
<span id="freezing-an-environment"></span><h3>4.6.2. Freezing an environment<a class="headerlink" href="#freezing-an-environment" title="Permalink to this headline">¶</a></h3>
<p>This will save the list of <strong>conda-installed</strong> software you have in a particular
environment to the file <code class="docutils literal notranslate"><span class="pre">packages.txt</span></code>:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="nb">list</span> <span class="o">--</span><span class="n">export</span> <span class="n">packages</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<p>(it will not record the software versions for software not installed by conda.)</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">install</span> <span class="o">--</span><span class="n">file</span><span class="o">=</span><span class="n">packages</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<p>will install those packages in your local environment.</p>
</div>
<div class="section" id="constructing-and-using-multiple-environments">
<span id="constructing-and-using-multiple-environments"></span><h3>4.6.3. Constructing and using multiple environments<a class="headerlink" href="#constructing-and-using-multiple-environments" title="Permalink to this headline">¶</a></h3>
<p>A feature that we do not use much here, but that can be very
handy in some circumstances, is different environments.</p>
<p>“Environments” are multiple different collections of installed software. There are two reasons you might want to do this:</p>
<ul class="simple">
<li>first, you might want to try to exactly replicate a specific software install, so that you can replicate a paper or an old condition.</li>
<li>second, you might be working with incompatible software, e.g. sometimes different software pipelines need different version of the same software. An example of this is older bioinformatics software that needs python2, while other software needs python3.</li>
</ul>
<p>To create a new environment named <code class="docutils literal notranslate"><span class="pre">pony</span></code>, type:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">create</span> <span class="o">-</span><span class="n">n</span> <span class="n">pony</span>
</pre></div>
</div>
<p>Then to activate (switch to) that environment, type:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">source</span> <span class="n">activate</span> <span class="n">pony</span>
</pre></div>
</div>
<p>And now when you run <code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">install</span></code>, it will install packages into this new environment, e.g.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">install</span> <span class="o">-</span><span class="n">y</span> <span class="n">checkm</span><span class="o">-</span><span class="n">genome</span>
</pre></div>
</div>
<p>(note here that checkm-genome <em>requires</em> python 2).</p>
<p>To list environments, type:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">env</span> <span class="nb">list</span>
</pre></div>
</div>
<p>and you will see that you have two environments, <code class="docutils literal notranslate"><span class="pre">base</span></code> and
<code class="docutils literal notranslate"><span class="pre">pony</span></code>, and <code class="docutils literal notranslate"><span class="pre">pony</span></code> has a <code class="docutils literal notranslate"><span class="pre">*</span></code> next to it because that’s your
current environment.</p>
<p>And finally, to switch back to your base environment, do:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">source</span> <span class="n">activate</span> <span class="n">base</span>
</pre></div>
</div>
<p>and you’ll be back in the original environment.</p>
</div>
<div class="section" id="meditations-on-reproducibility-and-provenance">
<span id="meditations-on-reproducibility-and-provenance"></span><h3>4.6.4. Meditations on reproducibility and provenance<a class="headerlink" href="#meditations-on-reproducibility-and-provenance" title="Permalink to this headline">¶</a></h3>
<p>If you want to impress reviewers and also keep track of
what your software versions are, you can:</p>
<ul class="simple">
<li>manage all your software inside of conda</li>
<li>use <code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">list</span> <span class="pre">--export</span> <span class="pre">software.txt</span></code> to create a list of all your software and put it in your supplementary material.</li>
</ul>
<p>This is also something that you can record for yourself, so that if you are trying to exactly reproduce</p>
</div>
<div class="section" id="using-it-on-your-own-compute-system-laptop-or-hpc">
<span id="using-it-on-your-own-compute-system-laptop-or-hpc"></span><h3>4.6.5. Using it on your own compute system (laptop or HPC)<a class="headerlink" href="#using-it-on-your-own-compute-system-laptop-or-hpc" title="Permalink to this headline">¶</a></h3>
<p>conda works on Windows, Mac, and Linux.</p>
<p>bioconda works on Mac and Linux.</p>
<p>It does not require admin privileges to install, so you can
install it on your own local cluster quite easily.</p>
</div>
</div>
<div class="section" id="bonus-installing-rstudio-web-on-your-running-linux-box">
<span id="bonus-installing-rstudio-web-on-your-running-linux-box"></span><h2>4.7. Bonus: installing RStudio Web on your running Linux box.<a class="headerlink" href="#bonus-installing-rstudio-web-on-your-running-linux-box" title="Permalink to this headline">¶</a></h2>
<p><strong>Note:</strong> this does require admin privileges, and you cannot run it on your local cluster. For your laptop, you can just install <a class="reference external" href="https://www.rstudio.com">the regular RStudio</a>.</p>
<hr class="docutils" />
<p>Install necessary system software (gdebi and R):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">sudo</span> <span class="n">apt</span><span class="o">-</span><span class="n">get</span> <span class="n">update</span> <span class="o">&amp;&amp;</span> <span class="n">sudo</span> <span class="n">apt</span><span class="o">-</span><span class="n">get</span> <span class="o">-</span><span class="n">y</span> <span class="n">install</span> <span class="n">gdebi</span><span class="o">-</span><span class="n">core</span> <span class="n">r</span><span class="o">-</span><span class="n">base</span>
</pre></div>
</div>
<p>Now, download and install RStudio:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">wget</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">download2</span><span class="o">.</span><span class="n">rstudio</span><span class="o">.</span><span class="n">org</span><span class="o">/</span><span class="n">rstudio</span><span class="o">-</span><span class="n">server</span><span class="o">-</span><span class="mf">1.1</span><span class="o">.</span><span class="mi">453</span><span class="o">-</span><span class="n">amd64</span><span class="o">.</span><span class="n">deb</span>
<span class="n">sudo</span> <span class="n">gdebi</span> <span class="o">-</span><span class="n">n</span> <span class="n">rstudio</span><span class="o">-</span><span class="n">server</span><span class="o">-</span><span class="mf">1.1</span><span class="o">.</span><span class="mi">453</span><span class="o">-</span><span class="n">amd64</span><span class="o">.</span><span class="n">deb</span>
</pre></div>
</div>
<p>At this point, RStudio will be running on port 8787, and you can <a class="reference external" href="https://angus.readthedocs.io/en/2018/visualizing-blast-scores-with-RStudio.html">follow these instructions</a> to set your password and log into it.</p>
</div>
</div>


          </div>
            
  <div class="footer-relations">
    
      <div class="pull-left">
        <a class="btn btn-default" href="jetstream/ssh_changepassword.html" title="previous chapter (use the left arrow)">3. Adding password to a Jetstream instance</a>
      </div>
    
    </div>
    <div class="clearer"></div>
  
        </div>
        <div class="clearfix"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="jetstream/ssh_changepassword.html" title="3. Adding password to a Jetstream instance"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">2018-setacna-rnaseq 0.0.1 documentation</a> &#187;</li> 
      </ul>
    </div>
<script type="text/javascript">
  $("#mobile-toggle a").click(function () {
    $("#left-column").toggle();
  });
</script>
<script type="text/javascript" src="_static/js/bootstrap.js"></script>
  <div class="footer">
    &copy; Copyright 2018, Elias Oziolor. Created using <a href="http://sphinx.pocoo.org/">Sphinx</a>.
  </div>
  </body>
</html>